content/en/docs/Creating derivatives/building_gce_images.md

```diff
@@ -0,0 +1,106 @@
+---
+title: "Build GCE images"
+linkTitle: "Build GCE images"
+weight: 4
+date: 2017-01-05
+description: >
+  This section documents the procedure to deploy cOS images
+  in GCE public cloud provider from scratch.
+---
+
+
+Requirements:
+
+* cOS-toolkit source
+* GCE access keys with the appropriate roles and permissions
+* [gsutil](https://cloud.google.com/storage/docs/gsutil) and [gcloud](https://cloud.google.com/sdk) tools
+
+The suggested approach high level view is building cOS packages and generating a RAW image from
+them. That would allow us to transform that RAW image in a valid GCE blob that can be transformed into a VM image ready
+to be launched.
+
+This generates a `vanilla` image that boots into [recovery mode](../../getting-started/recovery) and can be used to deploy
+whatever image you want to the VM. Then you can snapshot that VM into a VM image ready to deploy with the default cOS
+system or your derivative.
+
+## Building the packages
+
+We need to have the packages built locally in order to generate a proper RAW image. Just run:
+
+```bash
+sudo make build
+```
+
+All the artifacts will be generated under the `build` directory.
+
+
+See also the [dedicated section in the docs](../../development/) for more information about building cOS.
+
+## Building the RAW image
+
+The RAW image is just a RAW disk image that contains the recovery, so once launched is ready to be used for installing
+whatever cOS or derivative that you want into the VM disks. This allows us to have a barebones base image that can be
+used for provisioning whatever cOS you want.
+
+Building the RAW image is as simple as running:
+
+```bash
+sudo make raw_disk
+```
+
+This will output a `disk.raw` fiel in the current directory which we will modify to make it work with GCE.
+
+HINT: You can use this `disk.raw` locally as a disk for qemu if you want to check that it boots properly.
+
+## Transform the RAW image into a compatible GCE blob
+
+Currently importing images from storage blobs on GCE have a [few requirements](https://cloud.google.com/compute/docs/import/import-existing-image#requirements_for_the_image_file):
+
+ - blobs have to be tar.gzipped with the flag `--format=oldgnu`
+ - the disk.raw has to be rounded up to the next Gb ( so no 2.1gb images or 2.4, they need to be an exact 3Gb or 2Gb)
+ - image inside the tar.gzip blob needs to be called disk.raw
+
+We provide a make target that will do this for you:
+
+```bash
+sudo make gce_disk
+```
+
+This will create a `disk.raw.tar.gz` which is our final artifact
+
+### Uploading to GCE storage and importing it as an image
```

Comment:
I would move this section "Uploading to GCE" under Booting in getting started. I've added the instruction to import AWS images there as well because it sounded to me more consistent in order to explain on how to consume the artifacts we produce. In the long run, we are going to publish those within github releases, so it gives instruction on how to consume them.

I would add a link here to that section instead and also move this under `dev/` as it is documenting how to build `cOS` and not a derivative

Reply:
mmmh, I dont know. This is just for uploading the vanilla image, not instructions on how to boot it and configure with user-data. We may still add that to the booting part (although I dont really like it, it will end up really cluttered) but that should be a separate issue. I think in fact that it should be the other way around, booting page  linking to the aws page with the proper anchor, so all the cloud instructions for cloud X are in page X. If I want to know about how to do stuff in AWS I want to go to the AWS section and expect to find everything in there.

Reply:
having a separate section for the cloud sounds a good idea indeed. In my mind the story behind "I want to boot cOS up" should be linear, in the sense that it follows the user steps closely ( from Downloading , to Booting medium in a different way, and to customize later). That's also because 90% of our use-cases are, including later on to finalize the story with "how to create an image" with packer, depending on the cloud provider.

Reply:
To be fair, once we start publishing artifacts on releases, this will change a lot! probably this will be moved to under development as there would be not a need for anyone to do this, just download the artifact, upload, and create your image.

Maybe I should just add the same section for GCE in booting but using the artifacts from CI instead?

Reply:
btw, I dont remember what userdata I used in GCP I think so no idea the userdata needed :clown_face: 

Let me try to boot with our example user-data to see if it works, then update this  :D